lib/Driver/ToolChains/MSVC.cpp

```diff
@@ -1255,6 +1269,77 @@ void MSVCToolChain::AddClangCXXStdlibIncludeArgs(const ArgList &DriverArgs,
   // FIXME: There should probably be logic here to find libc++ on Windows.
 }
 
+void MSVCToolChain::addLinkerHelper(const ArgList &Args,
+                                    ArgStringList &CmdArgs,
+                                    bool IsLinker,
+                                    StringRef Arg) const {
+  if (IsLinker) {
+    CmdArgs.push_back("-linker");
+    CmdArgs.push_back(Args.MakeArgString(Arg));
+  } else {
+     CmdArgs.push_back(Args.MakeArgString(Arg));
+  }
+}
+
+void MSVCToolChain::AddFortranStdlibLibArgs(const ArgList &Args,
+                                            ArgStringList &CmdArgs,
+                                            bool AddMLinker) const {
+  bool staticFlangLibs = false;
+  bool useOpenMP = false;
+
+  const auto &D = getDriver();
+
+  if (Args.hasArg(options::OPT_staticFlangLibs)) {
+    for (auto *A: Args.filtered(options::OPT_staticFlangLibs)) {
+      A->claim();
+      staticFlangLibs = true;
+    }
+  }
+
+  Arg *A = Args.getLastArg(options::OPT_mp, options::OPT_nomp,
+                           options::OPT_fopenmp, options::OPT_fno_openmp);
+  if (A &&
+      (A->getOption().matches(options::OPT_mp) ||
+       A->getOption().matches(options::OPT_fopenmp))) {
+      useOpenMP = true;
+  }
+
+  if (!Args.hasArg(options::OPT_nostdlib, options::OPT_nostartfiles)) {
+     if (getDriver().IsFortranMode()) {
+       addLinkerHelper(Args, CmdArgs, AddMLinker, "/defaultlib:msvcrt");
+     } else if (!getDriver().IsCLMode() && !getDriver().IsFortranMode()) {
+       addLinkerHelper(Args, CmdArgs, AddMLinker, "/defaultlib:libcmt");
+     }
+  }
+
+  CmdArgs.push_back(Args.MakeArgString(std::string("-libpath:") +
```

Comment:
You need `if (!AddMLinker)` before this line. See https://github.com/flang-compiler/clang/pull/38/files#diff-51937622340a643fa06e18aa701d0837R705